nixos/tests/matrix/synapse-workers.nix

```
@@ -0,0 +1,45 @@
+  import ../make-test-python.nix ({ pkgs, ... }:
+  {
+    name = "matrix-synapse-workers";
+    meta = {
+    maintainers = pkgs.matrix-synapse.meta.maintainers;
+    };
+
+    nodes = {
+      homeserver = { pkgs, nodes, ... }: {
+        services.postgresql = {
+          enable = true;
+          initialScript = pkgs.writeText "synapse-init.sql" ''
+            CREATE ROLE "matrix-synapse" WITH LOGIN PASSWORD 'synapse';
+            CREATE DATABASE "matrix-synapse" WITH OWNER "matrix-synapse"
+            TEMPLATE template0
+            LC_COLLATE = "C"
+            LC_CTYPE = "C";
+          '';
+        };
+
+        services.redis.servers."".enable = true;
```

This is the old redis server layout and should no longer be used.

---

This is the new redis layout isn't it? With the new socket based redis connection stuff in synapse, this is no longer necessary. But before that, synapse was hardcoded to connect to redis without support for specifying which one.

AFAIK this was the only way to do it on nixos. Once I add the options for specifying redis socket I'll update this

---

I was not quite correct, it is the renamed option from the old ones, we should name the redis instance. https://github.com/NixOS/nixpkgs/blob/67ecd3fd9d46c3e4a29e0e5712ec1f618b140b02/nixos/modules/services/databases/redis.nix#L29-L49

```suggestion
        services.redis.servers."matrix-synapse".enable = true;
```

---

I agree, it's just that this requires more setup on the synapse side since it was hardcoded to talk to redis over tcp on `localhost:6379` which is exactly what the `""` named redis server is. But yes I will change this when implementing the socket options